Finalize adding Kuzmin-Kutuzov potential

jobovy opened the pull request:
@wilmatrick

I added a few more tests and tweaked the orbit-integration in Python and C test (which seems fine to me, the differences in z and vz are just somewhat bigger than those in R,vR, and vT because the values are typically smaller). I will merge this into the dev-v1.1 branch and then soon merge everything into master and release a new galpy version.

Thanks!

Jo

wilmatrick commented:
Wow! You added a LOT of tests. Thanks for this.
I'm glad, we finally have finished it. :-)

> @wilmatrick
>
> I added a few more tests and tweaked the orbit-integration in Python and C
> test (which seems fine to me, the differences in z and vz are just
> somewhat bigger than those in R,vR, and vT because the values are
> typically smaller). I will merge this into the dev-v1.1 branch and then
> soon merge everything into master and release a new galpy version.
>
> Thanks!
>
> Jo
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/jobovy/galpy/pull/241
>
> -- Commit Summary --
>
>   * Added file KuzminKutuzovStaeckelPotentia.py
>   * Added file KuzminKutuzovStaeckelPotential.c
>   * Added conversion Rz_to_lambdanu, corrected some typos.
>   * Replaced function _Rz_to_ln with Rz_to_lambdanu.
>   * Added conversion lambdanu_to_Rz.
>   * Removed _ln_to_Rz(), corrected typo.
>   * Replaced _dldR(),_dldz(),_dndR(),_dndz() with Rz_to_lambdanu_jac().
>   * Inserted Rz_to_lambdanu_jac() in the KuzminKutuzovStaeckel functions.
>   * Added Rz_to_lambdanu_hess().
>   * Inserted Rz_to_lambdanu_hess() in the KuzminKutuzovStaeckel functions.
>   * Added KuzminKutuzovStaeckelPotential c functions to galpy_potentials.h
>   * Added KuzminKutuzovStaeckelPotential to integrateFullOrbit.c ->
> parse_leapFuncArgs_Full()
>   * Added KuzminKutuzovStaeckelPotential to integrateFullOrbit.py ->
> _parse_pot()
>   * Added KuzminKutuzovStaeckelPotential to actionAngle.c ->
> parse_actionAngleArgs()
>   * Added the glue to use KuzminKutuzovStaeckelPotential C code for planar
> Orbit integration.
>   * Added amp in output of KuzminKutuzovStaeckelPotential c functions.
>   * Corrected some typos in KuzminKutuzovStaeckelPotential C code.
>   * Added KKSPot to potential.py, and corrected some minor issues in
> KKSPot and the coordinate transformation.
>   * Changed KKPot attribute from ac & Delta to _ac and _Delta.
>   * Corrected wrong indent.
>   * Merge branch 'dev-v1.1' of https://github.com/wilmatrick/galpy into
> dev-v1.1
>   * Corrected typo in a comment.
>   * Corrected typo in KKSPot nlderiv --> lnderiv.
>   * Made some KKSPot functions accessible for vector input.
>   * Replaced my calculations of the lambda, nu <--> R,z conversion
> Jacobian and Hessian with expressions derived with Mathematic. But both
> seem to be same.
>   * Merge MN3 conflicts
>   * fix faulty merge
>   * change tolerance for Kuzmin...
>   * try installing specific version of nose
>   * fix python3 import
>   * Changed fitting range in qdf.vmomentdensity from 4*sigma to
> nsigma*sigma, i.e. using the existing keyword.
>   * Removed Mathematica calculations from KKSpot and coordinate
> transformations again. Is the same as my calculations anyway.
>   * Merge pull request #235 from wilmatrick/dev-v1.1
>   * Added pre-factor for Gauss-Legendre intgration in qdf.vmomentdensity
> with flexible nsigma.
>   * Merge pull request #236 from wilmatrick/dev-v1.1
>   * add Kuzmin-Kutuzov to documentation and slight tweak to formatting of
> the Kuzmin documentation [ci skip]
>   * Added a test suite for the KuzminKutuzovStaeckelPotential.
>   * Added a test suite for the KuzminKutuzovStaeckelPotential. (Now with
> file.)
>   * Corrected some typos in the KKSPot test suite. Not finished yet...
>   * Merge remote-tracking branch 'origin/dev-v1.1' into wilma
>   * Improved array input treatment in KKSpot.
>   * Improved array input treatment in Jacobian/Hessian for coordinate
> transformations needed for KKSpot.
>   * Had to remove previous 'improved' array treatment to avoid the orbit
> integration to crash.
>   * Had to remove previous 'improved' array treatment to avoid the orbit
> integration to crash.
>   * Updated the test_kuzminkutuzov.py test suite. Works unfortunately only
> with 'ugly' array treatment in the KKS pot.
>   * KKSpot now supports array without explicit array treatment, and the
> scalar input in the Jacobian/Hessian of the KKSpot coordinate
> transformation is also accounted for. With this, the orbit integration
> works as well.
>   * Merge pull request #237 from wilmatrick/dev-v1.1
>   * add explicit test of Rz_to_lambdanu jacobian
>   * add explicit test of Rz_to_lambdanu hessian
>   * add special case of lambdanu_to_Rz test
>   * slight tweak to the C/Python orbit integration test for KuzminKutuzov,
> but seems to work fine
>
> -- File Changes --
>
>     M doc/source/reference/potential.rst (1)
>     A doc/source/reference/potentialkuzminkutuzov.rst (5)
>     M galpy/actionAngle_src/actionAngle_c_ext/actionAngle.c (6)
>     M galpy/df_src/quasiisothermaldf.py (20)
>     M galpy/orbit_src/integrateFullOrbit.py (3)
>     M galpy/orbit_src/integratePlanarOrbit.py (4)
>     M galpy/orbit_src/orbit_c_ext/integrateFullOrbit.c (7)
>     M galpy/orbit_src/orbit_c_ext/integratePlanarOrbit.c (8)
>     M galpy/potential.py (3)
>     A galpy/potential_src/KuzminKutuzovStaeckelPotential.py (300)
>     A galpy/potential_src/potential_c_ext/KuzminKutuzovStaeckelPotential.c
> (105)
>     M galpy/potential_src/potential_c_ext/galpy_potentials.h (11)
>     M galpy/util/bovy_coords.py (197)
>     M nose/test_coords.py (4)
>     A nose/test_kuzminkutuzov.py (482)
>     M nose/test_orbit.py (1)
>
> -- Patch Links --
>
> https://github.com/jobovy/galpy/pull/241.patch
> https://github.com/jobovy/galpy/pull/241.diff
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jobovy/galpy/pull/241
>

jobovy commented:
I only added two more tests, the commits listed are mostly yours :-)!

Thanks,

Jo

On Thu, Jun 11, 2015 at 9:42 AM, Wilma Trick <notifications@github.com>
wrote:

> Wow! You added a LOT of tests. Thanks for this.
> I'm glad, we finally have finished it. :-)
>
> > @wilmatrick
> >
> > I added a few more tests and tweaked the orbit-integration in Python and
> C
> > test (which seems fine to me, the differences in z and vz are just
> > somewhat bigger than those in R,vR, and vT because the values are
> > typically smaller). I will merge this into the dev-v1.1 branch and then
> > soon merge everything into master and release a new galpy version.
> >
> > Thanks!
> >
> > Jo
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/jobovy/galpy/pull/241
> >
> > -- Commit Summary --
> >
> > * Added file KuzminKutuzovStaeckelPotentia.py
> > * Added file KuzminKutuzovStaeckelPotential.c
> > * Added conversion Rz_to_lambdanu, corrected some typos.
> > * Replaced function _Rz_to_ln with Rz_to_lambdanu.
> > * Added conversion lambdanu_to_Rz.
> > * Removed _ln_to_Rz(), corrected typo.
> > * Replaced _dldR(),_dldz(),_dndR(),_dndz() with Rz_to_lambdanu_jac().
> > * Inserted Rz_to_lambdanu_jac() in the KuzminKutuzovStaeckel functions.
> > * Added Rz_to_lambdanu_hess().
> > * Inserted Rz_to_lambdanu_hess() in the KuzminKutuzovStaeckel functions.
> > * Added KuzminKutuzovStaeckelPotential c functions to galpy_potentials.h
> > * Added KuzminKutuzovStaeckelPotential to integrateFullOrbit.c ->
> > parse_leapFuncArgs_Full()
> > * Added KuzminKutuzovStaeckelPotential to integrateFullOrbit.py ->
> > _parse_pot()
> > * Added KuzminKutuzovStaeckelPotential to actionAngle.c ->
> > parse_actionAngleArgs()
> > * Added the glue to use KuzminKutuzovStaeckelPotential C code for planar
> > Orbit integration.
> > * Added amp in output of KuzminKutuzovStaeckelPotential c functions.
> > * Corrected some typos in KuzminKutuzovStaeckelPotential C code.
> > * Added KKSPot to potential.py, and corrected some minor issues in
> > KKSPot and the coordinate transformation.
> > * Changed KKPot attribute from ac & Delta to _ac and _Delta.
> > * Corrected wrong indent.
> > * Merge branch 'dev-v1.1' of https://github.com/wilmatrick/galpy into
> > dev-v1.1
> > * Corrected typo in a comment.
> > * Corrected typo in KKSPot nlderiv --> lnderiv.
> > * Made some KKSPot functions accessible for vector input.
> > * Replaced my calculations of the lambda, nu <--> R,z conversion
> > Jacobian and Hessian with expressions derived with Mathematic. But both
> > seem to be same.
> > * Merge MN3 conflicts
> > * fix faulty merge
> > * change tolerance for Kuzmin...
> > * try installing specific version of nose
> > * fix python3 import
> > * Changed fitting range in qdf.vmomentdensity from 4*sigma to
> > nsigma*sigma, i.e. using the existing keyword.
> > * Removed Mathematica calculations from KKSpot and coordinate
> > transformations again. Is the same as my calculations anyway.
> > * Merge pull request #235 from wilmatrick/dev-v1.1
> > * Added pre-factor for Gauss-Legendre intgration in qdf.vmomentdensity
> > with flexible nsigma.
> > * Merge pull request #236 from wilmatrick/dev-v1.1
> > * add Kuzmin-Kutuzov to documentation and slight tweak to formatting of
> > the Kuzmin documentation [ci skip]
> > * Added a test suite for the KuzminKutuzovStaeckelPotential.
> > * Added a test suite for the KuzminKutuzovStaeckelPotential. (Now with
> > file.)
> > * Corrected some typos in the KKSPot test suite. Not finished yet...
> > * Merge remote-tracking branch 'origin/dev-v1.1' into wilma
> > * Improved array input treatment in KKSpot.
> > * Improved array input treatment in Jacobian/Hessian for coordinate
> > transformations needed for KKSpot.
> > * Had to remove previous 'improved' array treatment to avoid the orbit
> > integration to crash.
> > * Had to remove previous 'improved' array treatment to avoid the orbit
> > integration to crash.
> > * Updated the test_kuzminkutuzov.py test suite. Works unfortunately only
> > with 'ugly' array treatment in the KKS pot.
> > * KKSpot now supports array without explicit array treatment, and the
> > scalar input in the Jacobian/Hessian of the KKSpot coordinate
> > transformation is also accounted for. With this, the orbit integration
> > works as well.
> > * Merge pull request #237 from wilmatrick/dev-v1.1
> > * add explicit test of Rz_to_lambdanu jacobian
> > * add explicit test of Rz_to_lambdanu hessian
> > * add special case of lambdanu_to_Rz test
> > * slight tweak to the C/Python orbit integration test for KuzminKutuzov,
> > but seems to work fine
> >
> > -- File Changes --
> >
> > M doc/source/reference/potential.rst (1)
> > A doc/source/reference/potentialkuzminkutuzov.rst (5)
> > M galpy/actionAngle_src/actionAngle_c_ext/actionAngle.c (6)
> > M galpy/df_src/quasiisothermaldf.py (20)
> > M galpy/orbit_src/integrateFullOrbit.py (3)
> > M galpy/orbit_src/integratePlanarOrbit.py (4)
> > M galpy/orbit_src/orbit_c_ext/integrateFullOrbit.c (7)
> > M galpy/orbit_src/orbit_c_ext/integratePlanarOrbit.c (8)
> > M galpy/potential.py (3)
> > A galpy/potential_src/KuzminKutuzovStaeckelPotential.py (300)
> > A galpy/potential_src/potential_c_ext/KuzminKutuzovStaeckelPotential.c
> > (105)
> > M galpy/potential_src/potential_c_ext/galpy_potentials.h (11)
> > M galpy/util/bovy_coords.py (197)
> > M nose/test_coords.py (4)
> > A nose/test_kuzminkutuzov.py (482)
> > M nose/test_orbit.py (1)
> >
> > -- Patch Links --
> >
> > https://github.com/jobovy/galpy/pull/241.patch
> > https://github.com/jobovy/galpy/pull/241.diff
> >
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/jobovy/galpy/pull/241
>
> >
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jobovy/galpy/pull/241#issuecomment-111139412>.
>
